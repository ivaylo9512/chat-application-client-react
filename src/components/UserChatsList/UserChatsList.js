import { useSelector } from "react-redux"
import { getCurrentUserChats } from "app/slices/userChatsSlice"
import UserChat from "components/UserChat/UserChat";
import { Container } from './UserChatsListStyle';

const UserChatsList = () => {
    const userChats = useSelector(getCurrentUserChats);

    return(
        <Container>
            {userChats && (userChats.length == 0
                ? <span data-testid='info'><b>No chats found with this search.</b></span> 
                : userChats.map(userChat => 
                    <UserChat key={userChat.id} userChat={userChat} />)
            )}
        </Container>
    )

}
export default UserChatsList