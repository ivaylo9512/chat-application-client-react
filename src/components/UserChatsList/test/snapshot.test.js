import * as redux from 'react-redux';
import { shallow } from 'enzyme';
import UserChatsList from 'components/UserChatsList/UserChatsList';

describe('UserChatsList snapshot tests', () => {
    const createWrapper = (state) => {
        jest.spyOn(redux, 'useSelector').mockReturnValue(state);

        return shallow(<UserChatsList />)
    }

    it('should render with users', () => {
        const wrapper = createWrapper([{id: 1, firstName: 'test'}, {id: 2, firstName: 'test'}, {id: 3, firstName: 'test'}, {id: 4, firstName: 'test'}, {id: 5, firstName: 'test'}])

        expect(wrapper).toMatchSnapshot();
    })

    it('should render info message when empty users array', () => {
        const wrapper = createWrapper([])
    
        expect(wrapper).toMatchSnapshot();
    })

    it('should not render container with undefined state', () => {
        const wrapper = createWrapper(undefined);

        expect(wrapper).toMatchSnapshot();
    })
})