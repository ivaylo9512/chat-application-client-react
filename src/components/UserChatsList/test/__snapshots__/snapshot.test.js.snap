// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserChatsList snapshot tests should not render container with undefined state 1`] = `<styled.div />`;

exports[`UserChatsList snapshot tests should render info message when empty users array 1`] = `
<styled.div>
  <span
    data-testid="info"
  >
    <b>
      No chats found with this search.
    </b>
  </span>
</styled.div>
`;

exports[`UserChatsList snapshot tests should render with users 1`] = `
<styled.div>
  <UserChat
    key="1"
    userChat={
      Object {
        "firstName": "test",
        "id": 1,
      }
    }
  />
  <UserChat
    key="2"
    userChat={
      Object {
        "firstName": "test",
        "id": 2,
      }
    }
  />
  <UserChat
    key="3"
    userChat={
      Object {
        "firstName": "test",
        "id": 3,
      }
    }
  />
  <UserChat
    key="4"
    userChat={
      Object {
        "firstName": "test",
        "id": 4,
      }
    }
  />
  <UserChat
    key="5"
    userChat={
      Object {
        "firstName": "test",
        "id": 5,
      }
    }
  />
</styled.div>
`;
